pkg/status/ptpconfig_status.go

```diff
@@ -0,0 +1,52 @@
+package status
+
+import (
+    "context"
+    "time"
+
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    "k8s.io/apimachinery/pkg/util/wait"
+
+    ptpv1 "github.com/k8snetworkplumbingwg/ptp-operator/api/v1"
+    clientset "github.com/k8snetworkplumbingwg/ptp-operator/pkg/client/clientset/versioned"
+)
+
+// UpsertPtpCondition updates or appends a condition on the PtpConfig status and issues UpdateStatus with retry
+func UpsertPtpCondition(ctx context.Context, cs clientset.Interface, namespace, name string, cond ptpv1.PtpCondition) error {
+    backoff := wait.Backoff{Duration: 100 * time.Millisecond, Factor: 1.5, Jitter: 0.1, Steps: 6}
+    return wait.ExponentialBackoff(backoff, func() (bool, error) {
+        pc, err := cs.PtpV1().PtpConfigs(namespace).Get(ctx, name, metav1.GetOptions{})
+        if err != nil {
+            return false, err
+        }
+
+        // upsert by Type+Profile
+        updated := false
+        for i := range pc.Status.PtpStatus.Conditions {
+            c := &pc.Status.PtpStatus.Conditions[i]
+            if c.Type == cond.Type && c.Profile == cond.Profile {
+                *c = cond
+                updated = true
+                break
+            }
+        }
+        if !updated {
+            pc.Status.PtpStatus.Conditions = append(pc.Status.PtpStatus.Conditions, cond)
+        }
+
+        // enforce max history
+        const maxConditions = 100
+        if n := len(pc.Status.PtpStatus.Conditions); n > maxConditions {
+            pc.Status.PtpStatus.Conditions = pc.Status.PtpStatus.Conditions[n-maxConditions:]
```

Comment:
So if one process (e.g. ptp4l) has too many statuses, we won't see detail from other processes such as ts2phc if it didn't report more recently than the last 100 ptp4l?